analyzers/nodejs/nodejs.go

```diff
@@ -218,40 +207,48 @@ func (a *Analyzer) IsBuilt() (bool, error) {
 func (a *Analyzer) Analyze() (graph.Deps, error) {
 	log.Debugf("Running Nodejs analysis: %#v", a.Module)
 
-	pkgs, err := a.Tool.List(filepath.Dir(a.Module.BuildTarget))
-	if err != nil {
-		log.Warnf("NPM had non-zero exit code: %s", err.Error())
-		log.Debug("Using fallback of node_modules")
+	pkgs, err := a.NPMTool.List(filepath.Dir(a.Module.BuildTarget))
+	if err == nil {
```

Comment:
restructuring checks for `err == nil` allows this func to avoid a level of nesting every time we use a fallback